crates/trigger-http/src/instrument.rs

```diff
@@ -0,0 +1,69 @@
+use anyhow::Result;
+use http::Response;
+use tracing::Level;
+use wasmtime_wasi_http::body::HyperIncomingBody;
+
+/// TODO
+macro_rules! http_span {
+    ($request:tt, $addr:tt) => {
+        tracing::info_span!(
+            "spin_trigger_http.handle_http_request",
+            "otel.kind" = "server",
+            "http.request.method" = %$request.method(),
+            "network.peer.address" = %$addr.ip(),
+            "network.peer.port" = %$addr.port(),
+            "network.protocol.name" = "http",
+            "url.path" = $request.uri().path(),
+            "url.query" = $request.uri().query().unwrap_or(""),
+            "url.scheme" = $request.uri().scheme_str().unwrap_or(""),
+            "client.address" = $request.headers().get("x-forwarded-for").and_then(|val| val.to_str().ok()),
+            // Recorded later
+            "error.type" = Empty,
+            "http.response.status_code" = Empty,
+            "http.route" = Empty,
+            "otel.name" = Empty,
+        )
+    };
+}
+
+pub(crate) use http_span;
+
+/// TODO
+pub struct RoutedResponse(pub Response<HyperIncomingBody>, pub Option<String>);
```

Comment:
I do think it should be possible (and preferable) to put the route info into a response extension.

Reply:
I'll give that a try. I guess I only tried putting it in the request extensions which had lifetime issues.

Are response extensions scoped to each specific response or are they global? We're going to run into issues if it's global.

Reply:
Extensions are scoped to the response instance. They're keyed by the extension value's [`TypeId`](https://doc.rust-lang.org/std/any/struct.TypeId.html) so you'll want a unique type e.g. `struct HttpRoute(String)`.

Reply:
If we don't go the extension route, then I would at the very least give the fields names instead of using a tuple struct.

Reply:
Resolving this b/c I got the extension route working.